frontends/web/src/components/transactions/components/tx-detail-dialog/tx-detail-dialog.tsx

```diff
@@ -0,0 +1,183 @@
+import { A } from '@/components/anchor/anchor';
+import { Dialog } from '@/components/dialog/dialog';
+import { Note } from './note';
+import { AmountWithUnit } from '@/components/amount/amount-with-unit';
+import { getTxSign } from '@/utils/transaction';
+import { ExternalLink } from '@/components/icon';
+import { TxDetailHeader } from '@/components/transactions/components/tx-detail-dialog/tx-detail-header';
+import { AdvancedTxDetail } from '@/components/transactions/components/tx-detail-dialog/advanced-tx-detail';
+import { TxDetailRow } from '@/components/transactions/components/tx-detail-dialog/tx-detail-row';
+import { AddressOrTxId } from '@/components/transactions/components/tx-detail-dialog/address-or-tx-id';
+import styles from './tx-detail-dialog.module.css';
+
+type TProps = {
+  open: boolean;
+  onClose: () => void;
+  accountCode: string;
+  internalID: string;
+  note: string;
+  status: TTransactionStatus;
+  type: TTransactionType;
+  numConfirmations: number;
+  numConfirmationsComplete: number;
+  time: string | null;
+  amount: TAmountWithConversions;
+  explorerURL: string;
+};
+
+export const TxDetailsDialogNew = ({
+  open,
+  onClose,
+  accountCode,
+  internalID,
+  note,
+  status,
+  type,
+  numConfirmations,
+  numConfirmationsComplete,
+  time,
+  amount,
+  explorerURL,
+}: TProps) => {
+  const { t } = useTranslation();
+  const [transactionInfo, setTransactionInfo] = useState<TTransaction | null>(null);
+
+  useEffect(() => {
+    if (!transactionInfo && open) {
+      getTransaction(accountCode, internalID).then(transaction => {
+        if (!transaction) {
+          console.error(`Unable to retrieve transaction ${internalID}`);
+        }
+        setTransactionInfo(transaction);
+      }).catch(console.error);
+    }
+  }, [accountCode, internalID, open, transactionInfo]);
+
+  if (transactionInfo === null) {
+    return;
+  }
+
+  const sign = transactionInfo && getTxSign(transactionInfo.type);
+  const displayedSign = sign === '−' ? sign : '';
+
+  // Amount and Confirmations info are displayed using props data
+  // instead of transactionInfo because they are live updated.
+
+  return (
+    <Dialog
+      open={open && !!transactionInfo}
+      title={t('transaction.details.title')}
+      onClose={onClose}
+      slim
+      medium>
+      {transactionInfo && (
+        <div className={styles.container}>
+          <TxDetailHeader
+            status={status}
+            numConfirmations={numConfirmations}
+            numConfirmationsComplete={numConfirmationsComplete}
+            type={type}
+            amount={amount}
+            transactionInfo={transactionInfo}
+            time={time}
+            displayedSign={displayedSign}
+          />
+
+          <hr className={styles.separator} />
+
+          <Note
+            accountCode={accountCode}
+            internalID={internalID}
+            note={note}
+          />
+
+          {/* to or send address */}
+          {type && (
```

Comment:
nit: it looks like check for `type` is not needed (wasn't in old version and not needed for weblint)